File: app/src/main/java/jgappsandgames/smartreminderslite/tasks/checkpoint/CheckpointActivityInterface.java
package jgappsandgames.smartreminderslite.tasks.checkpoint;

import android.app.Activity;
import android.os.Bundle;
import android.text.TextWatcher;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import org.json.JSONException;
import org.json.JSONObject;

import jgappsandgames.smartreminderslite.R;
import jgappsandgames.smartreminderslite.utility.ActivityUtility;
import jgappsandgames.smartreminderssave.tasks.Task;

/**
 * CheckpointActivityInterface
 * Created by joshua on 11/29/2017.
 */
public abstract class CheckpointActivityInterface extends Activity implements TextWatcher, View.OnClickListener, View.OnLongClickListener{
    // Views
    protected EditText text_view;
    protected Button status_button;

    // Lifecycle Methods
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Set Content View
        setContentView(R.layout.activity_checkpoint);

        // Set Empty REturn Intent
        setResult(ActivityUtility.RESPONSE_NONE);

        // Find
        text_view = findViewById(R.id.text);
        status_button = findViewById(R.id.status);
    }
}

File: app/src/main/java/jgappsandgames/smartreminderslite/tasks/checkpoint/CheckpointActivityInterface.kt
package jgappsandgames.smartreminderslite.tasks.checkpoint

// Android
import android.app.Activity
import android.os.Bundle

// Views
import android.text.TextWatcher
import android.view.View
import android.widget.Button
import android.widget.EditText

// App
import jgappsandgames.smartreminderslite.R
import jgappsandgames.smartreminderslite.utility.ActivityUtility

/**
 * CheckpointActivityInterface
 * Created by joshua on 1/19/2018.
 */
abstract class CheckpointActivityInterface: Activity(), TextWatcher, View.OnClickListener, View.OnLongClickListener {
    // Views ---------------------------------------------------------------------------------------
    protected var textView: EditText? = null
    protected var statusButton: Button? = null
    protected var continueButton: Button? = null

    // Lifecycle Methods ---------------------------------------------------------------------------
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Set Content View
        setContentView(R.layout.activity_checkpoint)

        // Set Empty Return Intent
        setResult(ActivityUtility.RESPONSE_NONE)

        // Find
        textView = findViewById(R.id.text)
        statusButton = findViewById(R.id.status)
        continueButton = findViewById(R.id.con)
    }
}